Remove duplicate_dequantize_node and remove_extra_dequantize

danrecoskie opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83611
* #83459

Summary: removed duplicate_dequantize_node and remove_extra_dequantize

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestQuantizeFxModels

Reviewers: jerryzh168

Subscribers:

Tasks:

Tags:

Differential Revision: [D38841052](https://our.internmc.facebook.com/intern/diff/D38841052)

Commit:
<Replace this line with a title. Use 1 line only, 67 chars or less>

Summary: removed duplicate_dequantize_node and remove_extra_dequantize

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestQuantizeFxModels

Reviewers: jerryzh168

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/fx/_lower_to_native_backend.py:
```diff
@@ -952,8 +934,8 @@ def _lower_to_native_backend(
     special_pattern_replacement(model)
-    model = fold_weight(model, node_name_to_scope)
     model.graph.eliminate_dead_code()
+    model = fold_weight(model, node_name_to_scope)
```
maybe run `eliminate_dead_code` again after this, just to be safe

jerryzh168 reviewed (approved):
looks great, thanks!

jerryzh168 commented:
can you add more details to the summary describing what we did? or you can just link the issue

Commit:
Update on "Remove duplicate_dequantize_node and remove_extra_dequantize"


Summary: Removed duplicate_dequantize_node and remove_extra_dequantize. See https://github.com/pytorch/pytorch/issues/83108 for more details.

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestQuantizeFxModels

Reviewers: jerryzh168

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "Remove duplicate_dequantize_node and remove_extra_dequantize"


Summary: Removed duplicate_dequantize_node and remove_extra_dequantize. See https://github.com/pytorch/pytorch/issues/83108 for more details.

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestQuantizeFxModels

Reviewers: jerryzh168

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

Commit:
Update on "Remove duplicate_dequantize_node and remove_extra_dequantize"


Summary: removed duplicate_dequantize_node and remove_extra_dequantize

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
python test/test_quantization.py TestQuantizeFxModels

Reviewers: jerryzh168

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

danrecoskie commented:
@danrecoskie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38841052).

jerryzh168 commented on torch/ao/quantization/fx/_lower_to_native_backend.py:
```diff
@@ -952,6 +933,7 @@ def _lower_to_native_backend(
     special_pattern_replacement(model)
+    model.graph.eliminate_dead_code()
     model = fold_weight(model, node_name_to_scope)
     model.graph.eliminate_dead_code()
```
looks like `model.graph.eliminate_dead_code()` is missing after 937 here?

jerryzh168 replied:
this looks good, sorry I misread this code before

danrecoskie commented:
@pytorchbot merge
